dartagnan/src/main/java/com/dat3m/dartagnan/verification/model/event/DefaultEventModel.java

```diff
@@ -0,0 +1,85 @@
+package com.dat3m.dartagnan.verification.model.event;
+
+import com.dat3m.dartagnan.program.event.core.*;
+import com.dat3m.dartagnan.program.event.*;
+import com.dat3m.dartagnan.verification.model.ThreadModel;
+
+import static com.dat3m.dartagnan.program.event.Tag.*;
+
+
+public class DefaultEventModel implements EventModel {
+    protected final Event event;
+    protected final ThreadModel thread;
+    protected int id = -1;
+
+    public DefaultEventModel(Event event, ThreadModel thread, int id) {
+        this.event = event;
+        this.thread = thread;
+        this.id = id;
+
+        thread.addEvent(this);
+    }
+
+    @Override
+    public Event getEvent() {
+        return event;
+    }
+
+    @Override
+    public ThreadModel getThreadModel() {
+        return thread;
+    }
+
+    @Override
+    public int getId() {
+        return id;
+    }
+
+    @Override
+    public boolean isMemoryEvent() { return event instanceof MemoryEvent; }
+
+    @Override
+    public boolean isInit() { return event instanceof Init; }
+
+    @Override
+    public boolean isWrite() { return event instanceof Store; }
+
+    @Override
+    public boolean isVisible() { return event instanceof MemoryEvent || event instanceof GenericVisibleEvent; }
+
+    @Override
+    public boolean isJump() { return event instanceof CondJump; }
+
+    @Override
+    public boolean isAssert() { return event instanceof Assert; }
+
+    @Override
+    public boolean isLocal() { return event instanceof Local; }
+
+    @Override
+    public boolean isRegReader() { return event instanceof RegReader; }
+
+    @Override
+    public boolean isRegWriter() { return event instanceof RegWriter; }
+
```

Comment:
Since all these are implemented as `instanceof`, what is the benefit of having these methods? 

There is at least one downside in the `ExecutionModelManager` where there is cast which is not obviously safe if one does not know how `isJump()` is implemented
```
if (em.isJump()) {
  for (EventModel dep : ((CondJumpModel) em).getDependentEvents(executionModel)) {
    rg.add(new Edge(em.getId(), dep.getId()));
  }
}
```

Reply:
I agree, I would get rid of all of them that are unnecessary (I mentioned this before, but I only marked a subset of those methods cause there were too many).